llvm/lib/Target/ARM/ARMISelLowering.cpp

```diff
@@ -14079,6 +14014,48 @@ static SDValue PerformSubCSINCCombine(SDNode *N, SelectionDAG &DAG) {
                      CSINC.getOperand(3));
 }
 
+static bool isNegatedInteger(SDValue Op) {
+  return Op.getOpcode() == ISD::SUB && isNullConstant(Op.getOperand(0));
+}
+
+static SDValue getNegatedInteger(SDValue Op, SelectionDAG &DAG) {
```

Comment:
There is `SelectionDAG::getNegative()` that does exactly the same.

Reply:
Thank you!